sigma_delta.h

```diff
@@ -0,0 +1,72 @@
+/* 
+ sigma_delta.h - esp8266 sigma-delta source
+
+ Copyright (c) 2014 Ivan Grokhotkov. All rights reserved.
+ This file is part of the esp8266 core for Arduino environment.
+ 
+ This library is free software; you can redistribute it and/or
+ modify it under the terms of the GNU Lesser General Public
+ License as published by the Free Software Foundation; either
+ version 2.1 of the License, or (at your option) any later version.
+
+ This library is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ Lesser General Public License for more details.
+
+ You should have received a copy of the GNU Lesser General Public
+ License along with this library; if not, write to the Free Software
+ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+ 
+ /*
+/******************************************************************************
+ * Info Sigma delta module
+
+This module controls the esp8266 internal sigma delta source
+Each pin can be connected to the sigma delta source
+The target duty and frequency can be modified via the register GPIO_SIGMA_DELTA
+
+THE TARGET FREQUENCY IS DEFINED AS:
+
+FREQ = 80,000,000/prescaler * target /256  HZ,     0<target<128
+FREQ = 80,000,000/prescaler * (256-target) /256  HZ,     128<target<256
+target: duty ,0-255
+prescaler: clk_div,0-255
+so the target and prescale will both affect the freq.
+
+Usage :
+1. sigma_delta_enable() : activate the sigma delta source with default prescalar (0) & target (0)
+2. sigma_delta_attachPin(pin), any pin 0..15, TBC if gpio16 supports sigma-delta source
+     This will set the pin to NORMAL output mode (pinMode(pin,OUTPUT))
+3. sigma_delta_setPrescaler(uint8_t) : reduce the output frequencies
+4. sigma_delta_setTarget(uint8_t) : set the output signal duty cycle, duty cycle = target/256
+
+5. sigma_delta_detachPin(pin), this will revert the pin to NORMAL output mode & GPIO source. 
+The sigma delta source remains on until :
+6. sigma_delta_disable()
+
+*******************************************************************************/
+
+#ifndef SIGMA_DELTA_H
+#define SIGMA_DELTA_H
+
+#ifdef __cplusplus
```

Comment:
The function prototypes here are "extern "C""'d, but the actual function definitions aren't.  Any reason not to drop the "extern C" here?

Reply:
linking stage doesn't work without this. Maybe you know why?
/SigmaDeltaDemo.ino:16: undefined reference to `sigmaDeltaGetPrescaler()'
and the same for all other sigmaDelta functions

Reply:
I just noticed you've got the file named "sigma_delta.**c**" and not "sigma_delta.**cpp**".  Can you "git mv" the file, that'll remove the need for this construction, and allow the better checking of the CPP compiler to be used, too.

Reply:
Ignore my previous comment.  The other "core_esp8266_*" files are all .**c**.  So ignore the request to remove extern C, leave it there.  Sorry for the confusion!

Reply:
Hi @earlephilhower ,
I don't manage to get the style_check right on the travis build. Could you have a look what the issue is, please?
The other builds are ok now.

Reply:
Will do tonight.  I think it's just a matter of getting the examples in Arduino IDE format (i.e. 2 space indent, brackets in the right spot, etc.).

Reply:
Looks like your problem is DOS linefeeds.  Can you save it with Unix ones, that seems to be what IGRR's checker requires to work properly.  I can't do it for you as I don't have write permission to your source repo (nor should I, of course!).